### PR TITLE
[AJER][3372] DD Data

### PR DESCRIPTION
**Pull request by andrewrichards1**

### CODE REVIEW

File: app/connectors/PaymentsOrchestratorConnector.scala [hunk 1]
```diff
@@ -70,4 +71,14 @@ class PaymentsOrchestratorConnector @Inject() (
     httpClient.GET[Option[CustomerInformation]](getCustomerURL)
   }
 
+  def getDDData(vrn: Vrn)(implicit request: Request[_]): Future[Option[DirectDebitData]] = {
+
+    implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
```
Comment (pawelpanasewicz) [inline app/connectors/PaymentsOrchestratorConnector.scala]: Please replace this line with:
`import req.RequestSupport._`. There is the implicit converted you were looking for :)

Comment (pawelpanasewicz) [reply app/connectors/PaymentsOrchestratorConnector.scala]: And please import it on class level, so all connector methods will see it.

File: app/format/DesFormatter.scala [hunk 2]
```diff
@@ -50,6 +50,32 @@ class CutomerInformationFormatter @Inject() (addressFormater: AddressFormter) {
...
+
+  def getDDData(directDebitData: Option[DirectDebitData]): Option[BankDetails] = {
+
+    val ddDetail: Option[DirectDebitDetails] = for {
+      dd <- directDebitData
+      ddDetailsOption <- dd.directDebitDetails
+      ddDetails <- ddDetailsOption.headOption
+    } yield (ddDetails)
+
+    ddDetail match {
```
Comment (pawelpanasewicz) [inline app/format/DesFormatter.scala]: better:

```
ddDetail.map(detail => BankDetails(detail.accountHolderName, detail.accountNumber, detail.sortCode))
```

File: app/model/des/DirectDebitData.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package model.des
+
+import play.api.libs.json.{Json, OFormat}
+
+final case class DirectDebitData(directDebitDetails: Option[List[DirectDebitDetails]]) {
```
Comment (pawelpanasewicz) [inline app/model/des/DirectDebitData.scala]: Maybe we can redefine it as `List[...]` only? `Some(Nil)` is pretty much the same as `None` here.

File: app/views/manage_or_track.scala.html [hunk 4]
```diff
@@ -61,15 +60,15 @@ <h1 id="main-message">@{LangMessages.`Manage or track`.show}</h1>
    <fieldset id="manage">
     <div class="form-group @form.errors("manage").map { e => form-group-error }" >
      @views.html.partials.error_message_form_input("manage", form)
-    <!-- DD option goes here -->
     @{
-    if(showBankAccountOption) {
+    ddDetailsOption.map(ddDetails => {
```
Comment (pawelpanasewicz) [inline app/views/manage_or_track.scala.html]: Here there is no need for extra `{`, `}` in the map. (It's cosmetics only ;))

File: app/connectors/PaymentsOrchestratorConnector.scala (hunk 1 above)
Comment (andrewrichards1) [reply app/connectors/PaymentsOrchestratorConnector.scala]: Cheers

**Pull request merged**